tests/graph/test_basic.py

```diff
@@ -302,6 +302,14 @@ def test_eval(self):
             pickle.loads(pickle.dumps(self.w)), "_fn_cache"
         ), "temporary functions must not be serialized"
 
+    def test_eval_with_strings(self):
+        assert self.w.eval({"x": 1.0, "y": 2.0}) == 6.0
+        assert self.w.eval({self.z: 3}) == 6.0
+        assert hasattr(self.w, "_fn_cache"), "variable must have cache after eval"
+        assert not hasattr(
+            pickle.loads(pickle.dumps(self.w)), "_fn_cache"
+        ), "temporary functions must not be serialized"
+
```

Comment:
I don't think we need these checks. Instead more useful would be to include a mixed case like `.eval({"x": 1.0, y: 2.0})` and check the cases where the string does not match any variable name.

Finally we may want to add one test case with repeated variable names such as:
```python
x = pt.scalar()
x.name="x"
x = x + 1
x.name = "x"
y = pt.exp(x)
y.eval({"x": 0})
```
I think we should fail in those cases. Does `get_var_by_name` return all the instances with the name?

Reply:
Hi @ricardoV94 ,

Thank you for the review.

`get_var_by_name` returns all instances of the same name. I am choosing the first declared variable i.e. last variable from the list to replace it with the name key.